11-lorawan/11-lorawan.md

```diff
@@ -0,0 +1,194 @@
+## Goal: Verify LoRaWAN networking
+
+Task #01 - LoRaWAN example
+==========================
+
+### Description
+
+LoRaWAN example is fully functional
+
+### Testing procedure
+
+Build and flash the `examples/lorawan` application with the following command
+
+    $ make BOARD=<board> DEVEUI=<device eui> APPEUI=<application eui> APPKEY=<application key> -C examples/lorawan flash term
+
+_Note:_ The application can be tested on IoT-LAB with a device register on
+TheThingsNetwork backend: start an experiment with one of the st-lrwan1 boards
+and pass `IOTLAB_NODE=auto-ssh` to the make command.
+
+### Result
+
+With valid device EUI, application EUI and application key, the example
+application can join the backend. Messages are received on the backend every
+~20s.
+
+
+Task #02 - OTAA join procedure
+==============================
+
+### Description
+
+OTAA join procedure and uplink with datarates indexes 0, 3, 5 (others are optional)
+
+### Testing procedure
+
+- Build and flash the `tests/pkg_semtech-loramac`
+
+      $ make BOARD=<board> -C tests/pkg_semtech-loramac flash term
+
+- Configure the devive EUI, application EUI, application key
+
+      > loramac set deveui 0011223344556677
+      > loramac set appeui 0011223344556677
+      > loramac set devkey 00112233445566770011223344556677
+
+- Configure the desired datarate index
+
+      > loramac set dr <datarate index>
+
+- Join the network using OTAA
+
+      > loramac join otaa
+
+- Send confirmed messages on port 123
+
+      > loramac tx <message> cnf 123
+
+- Send unconfirmed messages on port 42
+
+      > loramac tx <message> uncnf 42
+
+- Reboot the node after each datarate test
+
+### Result
+
+All joins are successful, messages are sent with success and correctly received
+on the LoRaWAN backend (same datarate, same port).
+TX attempts within the dutycycle are skipped (very likely to happen with low
+datarates). Messages sent from the backend to the device are received.
```

Comment:
with higher DR (means lower spreading factor) the time on air is smaller. So, I guess it should be the opposite:

```
...(very likely to happen with high datarates)
```

Reply:
Ok, this is confusing, I meant low datarate indexes (DR0 for instance)

Reply:
Well, now I'm re-reading it, I think it's fine as it now. Low datarates, DR0, will have a longer duty-cycle, so one has to wait more between 2 TX, that's why TX attempts have higher chances to be skipped with lower DR.

Reply:
ah! you are right. I read it again and of course makes sense